File: Melody.Server/Requests/UploadAudioRequest.cs
using Melody.Shared.Enums;

namespace Melody.Server.Requests
{
    public class UploadAudioRequest
    {
        public string Title { get; set; }

        public string Author { get; set; }

        public IFormFile? File { get; set; }

        public AudioType Type { get; set; }

        public string? Description { get; set; }

        public string[] GenreIds { get; set; }

        public string? CollectionId { get; set; }
    }
}

File: melody.client/src/common/requests/UploadAudioRequest.ts
import { AudioType } from "../enums/AudioType";

export type UploadAudioRequest = {
    title: string;
    author: string;
    file: File | null;
    type: AudioType;
    description: string | undefined | null;
    genreIds: string[];
    collectionId: string | undefined | null;
    creatorId: string;
}